### PR TITLE
LPS-146533 [CP 2.0] Update Status on Activation Form Card - Analytics Cloud by users with Provisioning role

### PR DESCRIPTION
**Pull request by joanagrego**
### **How it was developed?**

Creating a dropdown in the Analytic's Cloud card status component for a staff user to set status from "In Progress" to "Active" after the onboarding is complete.
Only a staff user can access this possibility through the dropdown menu to change the status. To update the status, there are some conditions to create the Group ID field in the modal.

### CODE REVIEW

**Review by iagochaves — APPROVED** [review]
Awesome!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/styles/components/_analytics-clould-modal.scss [hunk 1]
```diff
@@ -0,0 +1,4 @@
+.cp-analytics-cloud-status-modal {
+	height: 354px !important;
+	width: 600px !important;
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/styles/components/_analytics-clould-modal.scss:3]: Theses `!important` are required?
As a good practice, should be good avoid use them whenever possible :)

https://github.com/liferay/liferay-frontend-projects/blob/master/guidelines/css/when_to_use_css_utility_classes.md#2-the-important-attribute

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/styles/components/_activation-keys.scss [hunk 2]
```diff
@@ -34,18 +34,22 @@
...
 		padding-right: 68px !important;
 	}
 }
+
+.dropdown button:focus {
+	box-shadow: none;
+}
+
+.cp-activation-status-container .card-body {
+	padding-right: 13px !important;
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/styles/components/_activation-keys.scss:54]: Same as discuss about `!important` usage.
Maybe we can change to use the `pr-3` class, if possible.

There is no problem with `1px` difference.

https://github.com/liferay/liferay-frontend-projects/blob/master/guidelines/css/when_to_use_css_utility_classes.md#2-the-important-attribute

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/AnalyticsCloud/index.js [hunk 3]
```diff
@@ -109,10 +153,58 @@ const ActivationStatusAnalyticsCloud = ({
...
+					accountSubscriptionGroup: {
+						activationStatus: STATUS_TAG_TYPE_NAMES.active,
+					},
+					id:
+						subscriptionGroupAnalyticsCloud?.accountSubscriptionGroupId,
+				},
+			}),
+			await updateAnalyticsCloudWorkspaceId(),
+		]);
+		setSubscriptionGroupActivationStatus(STATUS_TAG_TYPE_NAMES.active);
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/AnalyticsCloud/index.js:196]:
```suggestion
		]);

		setSubscriptionGroupActivationStatus(STATUS_TAG_TYPE_NAMES.active);
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/AnalyticsCloud/AnalyticsCloudStatusModal.js [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+	const handleOnConfirm = () => {
+		const errorMessageGroupId = isLowercaseAndNumbers(groupIdValue);
+
+		if (errorMessageGroupId) {
+			setHasError(errorMessageGroupId);
+
+			return;
+		}
+		updateCardStatus();
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/AnalyticsCloud/AnalyticsCloudStatusModal.js:36]:
```suggestion
		}

		updateCardStatus();
```

**Review by jairmedeiros — CHANGES_REQUESTED** [review]

**Review by joaosvassis — APPROVED** [review]
Nice!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/AnalyticsCloud/index.js (hunk 3 above)
Comment (joanagrego) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/AnalyticsCloud/index.js:196]: Ok

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/styles/components/_analytics-clould-modal.scss (hunk 1 above)
Comment (joanagrego) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/styles/components/_analytics-clould-modal.scss:3]: Thank's!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/AnalyticsCloud/AnalyticsCloudStatusModal.js (hunk 4 above)
Comment (joanagrego) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/AnalyticsCloud/AnalyticsCloudStatusModal.js:36]: OK

**Review by Lucas2691 — APPROVED** [review]
Nice Work!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/styles/components/_activation-keys.scss [hunk 5]
```diff
@@ -34,18 +34,22 @@
...
+}
```
[the remark this answers is not in the document]
Comment (joanagrego) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/styles/components/_activation-keys.scss]: Done!

**Review by jairmedeiros — APPROVED** [review]

**Comment by jairmedeiros** [comment]
ci:forward

**Comment by jairmedeiros** [comment]
ci:forward

**Pull request closed without merging**